docs/get-started/onnx/index.md

```diff
@@ -0,0 +1,238 @@
+ONNX and use the ONNX encoder to deploy a Jina Flow.
+
+````{info}
+Jina, onnx, torch and torchvision are required to run this example. You can install the packages using:
+
+```
+pip install 'jina>=2.7' onnx torch torchvision
+```
+````
+
+
+## Exporting to ONNX
+
+Let's fine-tune a ResNet18 on a customized CelebA dataset.
+[Download the dataset](https://static.jina.ai/celeba/celeba-img.zip) and decompress it to
+`'./img_align_celeba'`.
+
+```python
+import finetuner as ft
+import torchvision
+from docarray import DocumentArray
+
+
+data = DocumentArray.from_files('img_align_celeba/*.jpg')
+
+
+def preprocess(doc):
+    return (
+        doc.load_uri_to_image_tensor(224, 224)
+        .set_image_tensor_normalization()
+        .set_image_tensor_channel_axis(-1, 0)
+    )
+
+
+data.apply(preprocess)
+
+resnet = torchvision.models.resnet18(pretrained=True)
+
+tuned_model = ft.fit(
+    model=resnet,
+    train_data=data,
+    loss='TripletLoss',
+    epochs=20,
+    batch_size=128,
+    to_embedding_model=True,
+    input_size=(3, 224, 224),
+    layer_name='adaptiveavgpool2d_67',
+    freeze=False,
+)
+
+```
+
+We can now export the model to the ONNX format, using the `to_onnx` method provided
+by finetuner. The `onnx` package is required to use this functionality. Also, if
+you are using TensorFlow or PaddlePaddle you will need to install additional packages
+for the ONNX conversion. Specifically you will need 
+[tf2onnx](https://github.com/onnx/tensorflow-onnx) for TensorFlow and
+[paddle2onnx](https://github.com/PaddlePaddle/Paddle2ONNX) for PaddlePaddle.
+
+```bash
+pip install onnx
+pip install tf2onnx
+pip install paddle2onnx
+```
+
+Back to the ONNX conversion:
+
+```python
+from finetuner.tuner.onnx import to_onnx
+
+to_onnx(tuned_model, 'tuned-model.onnx', input_shape=[3, 224, 224], opset_version=13)
+```
+
+The ONNX exported model should be stored now at `tuned-model.onnx`.
+
+A helper function for validating the ONNX export is also provided:
+```python
+from finetuner.tuner.onnx import validate_onnx_export
+
+validate_onnx_export(tuned_model, 'tuned-model.onnx', input_shape=[3, 224, 224])
+```
+
+This function compares the outputs of the native model and its ONNX counterpart against
+the same random input. If the outputs differ an assertion error is raised.
+
+Now that we have our model exported and have verified that it has the same behaviour as the
+original, it's time to deploy it 🚀.
+
+
+## Deploying using the ONNX Encoder
+
+You have already finetune your own model and transfer it to onnx format. Let's start deploying in Jina flow. If you are not familiar with Jina-Hub or Jina flow, check this:
+[Use Hub Executor](https://docs.jina.ai/advanced/hub/use-hub-executor/)
+[Use Jina Flow](https://docs.jina.ai/fundamentals/flow/)
+
+Here are steps you can follow:
+
+### Add ONNXEncoder to Jina flow
+
+We already have onnx encoder on Jina-Hub, let's add ONNXEncoder to jina flow:
```

Comment:
We already have **an** **ONNX** encoder on Jina-Hub, let's add `ONNXEncoder` to jina flow: